packages/connect-migrate/src/lib/migrate-source-files.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Scanned } from "./scan";
+import j from "jscodeshift";
+import { Logger, PrintFn } from "./logger";
+import { updateSourceFile } from "./update-source-file";
+
+interface MigrateSourceFilesResult {
+  sourceFileErrors: number;
+}
+
+export function migrateSourceFiles(
+  scanned: Scanned,
+  transform: j.Transform,
+  print: PrintFn,
+  logger?: Logger,
+  updateSourceFileFn: typeof updateSourceFile = updateSourceFile,
+): MigrateSourceFilesResult {
```

Comment:
How do you feel about moving these to named arguments? It feels like there are enough options here (and likely more to be added) that naming would help.

Reply:
Pushed up https://github.com/connectrpc/connect-es/pull/787/commits/6b9428f81ab78d217f0e47a51c4bc6212ac9ab22 for improved readability in the test setup. I don't see similar improvements for the migrate* functions, so left them as is.

Reply:
Makes sense to me